springboot/src/main/java/ca/jrvs/apps/trading/dao/MarketDataDao.java

```diff
@@ -0,0 +1,209 @@
+        String url = String.format(IEX_BATCH_URL, tickerListString);
+        Optional<String> quoteString = null;
+        try {
+            quoteString = executeHttpGet(url);
+        } catch (Exception e) {
+            logger.error("Invalid url");
+            throw new DataRetrievalFailureException(e.getMessage());
+        }
+        JSONObject jsonObject = new JSONObject(quoteString.get());
+        ObjectMapper mapper = new ObjectMapper();
+        IexQuote quote;
+        for (String ticker : tickerList) {
+            String stringQuote = jsonObject.getJSONObject(ticker).getJSONObject("quote").toString();
+            try {
+                quote = mapper.readValue(stringQuote, IexQuote.class);
+            } catch (IOException ex) {
+                logger.error("Failed to convert JSON to quote object");
+                throw new JSONException(ex.getMessage());
+            }
+            quotes.add(quote);
+        }
+        return quotes;
+    }
+
+
+    /**
+     * Execute a get and return http entity/body as a string
+     * <p>
+     * Use EntityUtils.toString to process HTTP entity
+     *
+     * @param url resource URL
+     * @return HTTP response body or Optional.empty for 404 response
+     * @throws DataRetrievalFailureException if HTTP failed or status code is unexpected
+     */
+    private Optional<String> executeHttpGet(String url) throws DataRetrievalFailureException, MalformedURLException {
+        HttpClient httpClient = getHttpClient();
+        HttpGet httpRequest = new HttpGet(url);
+        HttpResponse httpResponse;
+        try {
+            httpResponse = httpClient.execute(httpRequest);
+        } catch (IOException e) {
+            logger.error("Invalid URL");
+            throw new MalformedURLException(e.getMessage());
+        }
+        int status = httpResponse.getStatusLine().getStatusCode();
+        if (status != HTTP_OK) {
+            throw new DataRetrievalFailureException("Unexpected http status" + status);
+        }
+        if (httpResponse.getEntity() == null) {
+            throw new RuntimeException("Empty response body");
+        }
+        String jsonString;
+        try {
+            HttpEntity httpEntity = httpResponse.getEntity();
+            jsonString = EntityUtils.toString(httpEntity);
+        } catch (IOException ex) {
+            logger.error("Conversion from entity to string failed");
+            throw new IllegalArgumentException(ex.getMessage());
+        }
+        Optional<String> response = Optional.of(jsonString);
+        return response;
+    }
+
+    /**
+     * Borrow a HTTP client from HttpClientConnectionManager
+     *
+     * @return httpClient
+     */
+
+    private CloseableHttpClient getHttpClient() {
+        return HttpClients.custom()
+                .setConnectionManager(httpClientConnectionManager)
+                .setConnectionManagerShared(true)
+                .build();
+    }
+
+    @Override
+    public long count() {
```

Comment:
For unimplemented, suggest to throw UnsupportedOperationException like example in notion

Reply:
Done, thanks!